app/helpers/lookup_keys_helper.rb

```diff
@@ -35,7 +35,7 @@ def show_puppet_class(f)
     # Show the parent puppetclass as a context, but permit no change.
     if params["puppetclass_id"]
       select_f f, :puppetclass_id, [Puppetclass.find(params["puppetclass_id"])], :id, :to_label, {}, {:label => _("Puppet class"), :disabled => true}
-    elsif f.object.is_param && f.object.param_class
+    elsif f.object.class == PuppetclassLookupKey && f.object.param_class
```

Comment:
DRY
this is used a lot, so put a method called "puppetclass?" on LookupKey which checks this.